libraries/ESP8266WiFi/src/WiFiClientSecure.cpp

```diff
@@ -564,6 +564,29 @@ size_t WiFiClientSecure::write_P(PGM_P buf, size_t size)
     return write(copy, size);
 }
 
+// The axTLS bare libs don't understand anything about Arduino Streams,
+// so we have to manually read and send individual chunks.
+size_t WiFiClientSecure::write(Stream& stream)
+{
+    size_t totalSent = 0;
+    size_t countRead;
+    size_t countSent;
+    if (!_ssl)
+    {
+        return 0;
+    }
+    do {
+        uint8_t temp[128]; // Arbitrary temporary chunk size
+        countSent = 0;
+        countRead = stream.readBytes(temp, sizeof(temp));
+        if (countRead) {
+            countSent = write(temp, countRead);
+            totalSent += countSent;
+        }
+    } while ( (countSent == countRead) && (countSent > 0) );
```

Comment:
What happens if the streaming takes a long time? wouldn't you get a wdt timeout?

Reply:
Good catch.  Just checked the ClientContext code and matched their yield and chunk sizes.  Tested against a 158KB HTTPS file transfer which took many seconds to complete.